src/main/java/com/softserve/repository/impl/LessonRepositoryImpl.java

```diff
@@ -320,21 +336,27 @@ public Integer updateLinkToMeeting(Lesson lesson) {
      * @return updated Lesson
      */
     @Override
-    public Lesson updateGrouped(Lesson oldLesson,Lesson updatedLesson) {
+    public Lesson updateGrouped(Lesson oldLesson,Lesson updatedLesson, boolean isTeacherOrSubjectUpdated) {
         log.info("Entered updateGroup({}, {})", oldLesson, updatedLesson);
-        int updated = sessionFactory.getCurrentSession()
-                .createQuery(UPDATE_GROUPED)
-                .setParameter("initialSubjectId", oldLesson.getSubject().getId())
+        Session currentSession = sessionFactory.getCurrentSession();
+        Query query = isTeacherOrSubjectUpdated
+                ? currentSession.createQuery(UPDATE_GROUPED_TEACHER_OR_SUBJECT)
+                : currentSession.createQuery(UPDATE_GROUPED);
+        query = query.setParameter("initialSubjectId", oldLesson.getSubject().getId())
                 .setParameter("initialHours", oldLesson.getHours())
                 .setParameter("initialTeacherId", oldLesson.getTeacher().getId())
                 .setParameter("initialSemesterId", oldLesson.getSemester().getId())
                 .setParameter("initialLessonType", oldLesson.getLessonType())
+                .setParameter("linkToMeeting", updatedLesson.getLinkToMeeting())
                 .setParameter("subjectId", updatedLesson.getSubject().getId())
                 .setParameter("hours", updatedLesson.getHours())
                 .setParameter("teacherId", updatedLesson.getTeacher().getId())
                 .setParameter("lessonType", updatedLesson.getLessonType())
-                .setParameter("subjectForSite", updatedLesson.getSubjectForSite())
-                .executeUpdate();
+                .setParameter("subjectForSite", updatedLesson.getSubjectForSite());
+        query = isTeacherOrSubjectUpdated
```

Comment:
This does seem odd to me, 2 different queries are being executed hinging on one parameter? How is it different from initialHours/initialLessonType?

Reply:
No difference. I need to delete initialHours and InitialLessonType from UPDATE_GROUPED_TEACHER_OR_SUBJECT